ext/drreg/drreg.c

```diff
@@ -215,10 +286,63 @@ spill_reg(void *drcontext, per_thread_t *pt, reg_id_t reg, uint slot, instrlist_
 #endif
 }
 
+#ifdef X86
+static void
+load_indirect_block(void *drcontext, per_thread_t *pt, uint slot, instrlist_t *ilist,
+                    instr_t *where, reg_id_t tmp_reg)
+{
+    LOG(drcontext, DR_LOG_ALL, 3, "%s @%d." PFX " %s %d\n", __FUNCTION__, pt->live_idx,
+        get_where_app_pc(where), get_register_name(tmp_reg), slot);
+    /* Simply load the pointer of the block to the passed register*/
+    dr_insert_read_raw_tls(drcontext, ilist, where, tls_seg, slot, tmp_reg);
+}
+#endif
+
+#ifdef X86
+static void
+spill_reg_indirectly(void *drcontext, per_thread_t *pt, reg_id_t reg, uint slot,
+                     instrlist_t *ilist, instr_t *where)
+{
+    reg_id_t tmp_reg = DR_REG_NULL;
```

Comment:
Any reason why you named the register tmp_reg? It seems a bit too generic and block_reg was more implicit in terms to use in relation readability. Perhaps tmp_block_reg?

Reply:
It's literally a temp. tmp_block_reg: In most places, we call it scratch reg, let's do scratch_block_reg, done.